account_invoice_show_currency_rate/models/account_move.py

```diff
@@ -26,29 +26,41 @@ class AccountMove(models.Model):
     def _compute_currency_rate_amount(self):
         """ It's necessary to define value according to some cases:
         - Case A: Currency is equal to company currency (Value = 1)
-        - Case B: Move exist previously (posted) and get real rate according to lines
-        - Case C: Get expected rate (according to date) to show some value in creation.
+        - Case B: Move exist previously (posted)
+        and get real rate according to lines
+        - Case C: Get expected rate (according to date)
+        to show some value in creation.
         """
-        self.currency_rate_amount = 1
-        for item in self.filtered("show_currency_rate_amount"):
-            lines = item.line_ids.filtered(lambda x: abs(x.amount_currency) > 0)
-            if item.state == "posted" and lines:
-                amount_currency_positive = sum(
-                    [abs(amc) for amc in lines.mapped("amount_currency")]
-                )
-                total_balance_positive = sum([abs(b) for b in lines.mapped("balance")])
-                item.currency_rate_amount = item.currency_id.round(
-                    amount_currency_positive / total_balance_positive
-                )
-            else:
-                rates = item.currency_id._get_rates(item.company_id, item.date)
-                item.currency_rate_amount = rates.get(item.currency_id.id)
+        for item in self:
+            item.currency_rate_amount = 1
+            if item.show_currency_rate_amount:
```

Comment:
`if not item.show_currency_rate_amount:
    continue
`
then it makes it more readable